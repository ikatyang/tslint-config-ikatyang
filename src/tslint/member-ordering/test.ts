// good

class good1 {
  static d1;
  static d2;
  static d3;
  static c1() {}
  static c2() {}
  static c3() {}
  b1;
  b2;
  b3;
  constructor() {}
  a1() {}
  a2() {}
  a3() {}
}

// bad

class bad1 {
  static d3;
  static d2;
  static d1;
  static c2() {}
  static c3() {}
  b1;
  b2;
  b3;
  constructor() {}
  a1() {}
  a2() {}
  a3() {}
  static c1() {}
}
