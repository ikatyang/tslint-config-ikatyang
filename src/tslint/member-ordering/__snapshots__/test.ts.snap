// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint 1`] = `
// good

class good1 {
  static d1;
  static d2;
  static d3;
  static c1() {}
  static c2() {}
  static c3() {}
  b1;
  b2;
  b3;
  constructor() {}
  a1() {}
  a2() {}
  a3() {}
}

// bad

class bad1 {
  static d3;
  static d2;
         ~~  ['d2' should come alphabetically before 'd3']
  static d1;
         ~~  ['d1' should come alphabetically before 'd3']
  static c2() {}
  static c3() {}
  b1;
  b2;
  b3;
  constructor() {}
  a1() {}
  a2() {}
  a3() {}
  static c1() {}
  ~~~~~~~~~~~~~~ [Declaration of static-method not allowed after declaration of instance-method. Instead, this should come after static-fields.]
}

`;
